### DISCUSSION TITLE
Adding Multiple Genre's

### DISCUSSION THREAD

**Discussion by ghost** [Zeugma440/atldotnet#271] [General]
Hello,
      When I try to a string with semi-colon-delimited genre's, it only writes one genre to the file:
      
      Track scannedTrack = new Track(song.url);
      scannedTrack.Genre = "Youtube Likes; Test; Test2;";

      
![Screenshot 2024-06-22 013805](https://github.com/Zeugma440/atldotnet/assets/30101991/29cce2c4-78b3-445a-ad30-be300a9a914b)

**Comment by ghost** [comment]
Guys, according to my brief, research, you can't have more than one Genre. Most suggest using Pipe symbol to delimit the genre's.

**Reply by Zeugma440** [reply]
Hi there~

You should separate multiple values within any field using `ATL.Settings.DisplayValueSeparator`. The library will handle the rest according to specs ;-)

**Reply by ghost** [reply]
Cool thanks a lot man I'll give this a try and post an update. I have seen multiple semicolon concatenated genres. While Windows won't let me set it through the properties window I think perhaps your advice might let me do it programmatically.

**Reply by Zeugma440** [reply]
If you really want to get into specifics, what specs say is :
- ID3v2.2 and ID3v2.3 => values are separated using `/`
- ID3v2.4 => values are separated using `\0`

I'm pretty sure the semicolon is an UI convention most softwares have adopted.

**Reply by ghost** [reply]
Hi,
    I really appreciate the tip. I'm still only getting the first string. Am I using the separator correctly?
    
    scannedTrack.Genre = originalGenre + ATL.Settings.DisplayValueSeparator + entry.genre;

originalGenre & entry.genre are of type String.

**Reply by Zeugma440** [reply]
- Are you still testing on an MP3 file?
- How do you test if it works? Are you using the Windows file explorer?

**Reply by ghost** [reply]
Hi,

Yes, it is an MP3 file. Using Winamp and File Explorer Properties window. I see that if I write the Track.Genre to the console, I see all the concatenated genre's. This is good. Is there a way to see more than one genre in the Windows Explorer properties window?

I have two goals:

1. Have the genres in the metadata for me to programmatically parse. It seems this has already been achieved. 

2. Have the genres available in a Windows/.NET application like Winamp, so I can filter using the genres.


I think I have some FLAC files that have more than one genre that Windows allows me to select individually in the properties window, just like tags for JPEGs. Should I just convert all my MP3's to FLAC to get the metadata I want in Windows?

**Reply by Zeugma440** [reply]
> Have the genres available in a Windows/.NET application like Winamp, so I can filter using the genres.

I've made a small test : I tagged an MP3 files with two genres using foobar2000 v2.1.5, and opened that file inside all players installed on my PC.

**Software that only display one single genre**
- Windows Explorer
- Winamp v5.8
- AIMP v4.51
- PotPlayer v220420
- iTunes v12.10.11.2

**Software that successfuly display both genres**
- Foobar2000 v2.1.5
- MusicBee v3.2.6902
- VLC v3.0.18

=> As you can see, many "real life" software have not been designed to handle more than one genre, although the ID3v2 specification allow it.

I guess you need to pick one of the three that does support multiple genres if you want to acheive your goal  ;-)

> I think I have some FLAC files that have more than one genre that Windows allows me to select individually in the properties window, just like tags for JPEGs. 

Could you be more specific? I don't see any difference between MP3s and FLACs when reading / editing their metadata with Windows Explorer.

> Should I just convert all my MP3's to FLAC to get the metadata I want in Windows?

Certainly not; that's a very bad idea, as you'd encode a lossy signal using a lossless codec, which is more or less heresy.
